core/src/main/java/me/aurium/branch/annotate/TypeToken.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *  Branch
+ *  Copyright © 2021 Aurium
+ *
+ *  Branch is free software: you can redistribute it and/or modify
+ *  It under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  Branch is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with Branch. If not, see <https://www.gnu.org/licenses/>
+ *  and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+package me.aurium.branch.annotate;
+
+public class TypeToken<T> {
```

Comment:
A class like this ought to be given more care. It should validate its constructor argument is non-null, be `final` to enforce immutability, and implement equals/hashCode/toString

Reply:
This hasn't been used anywhere - all of the annotation stuff is pretty much unworked on